web/app/js/components/ConfigureProfilesMsg.jsx

```diff
@@ -89,46 +117,75 @@ class ConfigureProfilesMsg extends React.Component {
             <DialogContentText>
-              To create a service profile, download a profile and then apply it with `kubectl apply`.
+              To create a service profile, download a profile and then apply it
+              with `kubectl apply`.
             </DialogContentText>
-            <TextField
-              id="service"
-              label="Service"
+            <FormControl
               className={classes.textField}
-              value={this.state.service}
-              onChange={this.handleChange('service')}
-              margin="normal" />
-            <TextField
-              id="namespace"
-              label="Namespace"
+              onBlur={() => this.validateFields('service', query.service)}
+              error={error.service}>
+              <InputLabel htmlFor="component-error">Service</InputLabel>
+              <Input
+                id="component-error"
+                value={this.state.service}
+                onChange={this.handleChange('service')}
+                aria-describedby="component-error-text" />
+              {error.service && (
+                <FormHelperText id="component-error-text">
+                  The Service name must consist of lower case alphanumeric
+                  characters or &#45; and must start and end with an
+                  alphanumeric character
+                </FormHelperText>
+              )}
+            </FormControl>
+            <FormControl
               className={classes.textField}
-              value={this.state.name}
-              onChange={this.handleChange('namespace')}
-              margin="normal" />
+              onBlur={() => this.validateFields('namespace', query.namespace)}
+              error={error.namespace}>
+              <InputLabel htmlFor="component-error">Namespace</InputLabel>
+              <Input
+                id="component-error"
+                value={this.state.name}
+                onChange={this.handleChange('namespace')}
+                aria-describedby="component-error-text" />
+              {error.namespace && (
+                <FormHelperText id="component-error-text">
+                  The Namespace name must consist of lower case alphanumeric
+                  characters or &#45; and must start and end with an
+                  alphanumeric character
+                </FormHelperText>
+              )}
+            </FormControl>
           </DialogContent>
-
           <DialogActions>
-            <Button onClick={this.handleClose} color="primary">Cancel</Button>
+            <Button onClick={this.handleClose} color="primary">
+              Cancel
+            </Button>
             <a href={downloadUrl} style={{ textDecoration: 'none' }}>
               <Button
-                disabled={_isEmpty(query.service) || _isEmpty(query.namespace)}
-                onClick={this.handleClose}
-                color="primary">Download
+                disabled={_isEmpty(query.service) ||
```

Comment:
even when there's an error and the Download button is greyed out, I can still click it (it's not truly disabled). This is because this button is wrapped in a link to provide the download functionality. You'll need to do something like the following to make it truly unclickable:


```diff
diff --git a/web/app/js/components/ConfigureProfilesMsg.jsx b/web/app/js/components/ConfigureProfilesMsg.jsx
index 53bbb4b3..626e2bcb 100644
--- a/web/app/js/components/ConfigureProfilesMsg.jsx
+++ b/web/app/js/components/ConfigureProfilesMsg.jsx
@@ -123,6 +123,16 @@ class ConfigureProfilesMsg extends React.Component {
       );
     }

+    let disableDownloadButton = _isEmpty(query.service) || _isEmpty(query.namespace) ||
+      error.service || error.namespace;
+    let downloadButton = (
+      <Button
+        disabled={disableDownloadButton}
+        onClick={() => this.handleClose(downloadUrl)}
+        color="primary">
+        Download
+      </Button>
+    );
     return (
       <React.Fragment>
         {button}
@@ -177,17 +187,13 @@ class ConfigureProfilesMsg extends React.Component {
             <Button onClick={this.handleClose} color="primary">
               Cancel
             </Button>
-            <a href={downloadUrl} style={{ textDecoration: 'none' }}>
-              <Button
-                disabled={_isEmpty(query.service) ||
-                  _isEmpty(query.namespace) ||
-                  error.service ||
-                  error.namespace}
-                onClick={() => this.handleClose(downloadUrl)}
-                color="primary">
-                Download
-              </Button>
-            </a>
+            {
+              disableDownloadButton ? downloadButton :
+              <a
+                href={disableDownloadButton ? '' : downloadUrl}
+                style={{ textDecoration: 'none' }}>{downloadButton}
+              </a>
+            }
           </DialogActions>
         </Dialog>
       </React.Fragment>
```